### PR TITLE
fix(docs): Fix auto-generated docs

### PR DESCRIPTION
**Pull request by peknur**
Use tfplugindocs to generate docs using examples and resource descriptions.
Added also a new Make target "docs" (make docs).

### CODE REVIEW

File: docs/resources/storage.md [hunk 1]
```diff
@@ -150,13 +129,6 @@ Required:
 - **retention** (Number) The number of days before a backup is automatically deleted
 - **time** (String) The time of day when the backup is created
 
-Notes:
```
Comment (ka-myl) [inline docs/resources/storage.md:153]: Is there some way of adding such notes to the output? Via comment or something? If not, then we should move this information to examples somehow I guess

**Comment by ka-myl** [comment]
Is there some way we can remove computed-only properties from the docs output? For example server resource docs specify that `id` is an optional property, when in fact user cannot set it. Would be cool to filter those out

File: docs/resources/storage.md (hunk 1 above)
Comment (peknur) [reply docs/resources/storage.md:153]: Missed those notes :) I created new template for server and added those notes to template.

**Comment by peknur** [comment]
> Is there some way we can remove computed-only properties from the docs output? For example server resource docs specify that `id` is an optional property, when in fact user cannot set it. Would be cool to filter those out

I think that we cannot do that at the moment with tfplugindocs.

**Review by ka-myl — APPROVED** [review]
LGTM

**Pull request merged**